rust/nasl-builtin-std/src/report_functions.rs

```diff
@@ -0,0 +1,130 @@
+use std::sync::{Arc, RwLock};
+
+use models::{Protocol, ResultType};
+use nasl_builtin_utils::{Context, ContextType, FunctionErrorKind, Register};
+use nasl_syntax::NaslValue;
+
+#[derive(Debug, Clone, Default)]
+/// The description builtin function
+pub struct Reporting {
+    id: Arc<RwLock<usize>>,
+}
+
+impl Reporting {
+    fn id(&self) -> usize {
+        let mut id = self.id.as_ref().write().expect("expected write lock");
+        let result = *id;
+        *id += 1;
+        result
+    }
+
+    fn store_result(
+        &self,
+        typus: ResultType,
+        register: &Register,
+        context: &Context,
+    ) -> Result<NaslValue, FunctionErrorKind> {
+        let data = register.named("data").map(|x| x.to_string());
+        let port = register
+            .named("port")
+            .and_then(|x| match x {
+                ContextType::Value(x) => Some(x.into()),
+                _ => None,
+            })
+            .map(|x: i64| x as i16);
+
+        let protocol = match register
+            .named("proto")
+            .map(|x| x.to_string())
+            .as_ref()
+            .map(|x| x as &str)
+        {
+            Some("udp") => Protocol::UDP,
+            _ => Protocol::TCP,
+        };
+        // TOO: enhance id
```

Comment:
```suggestion
        // TODO: enhance id
```
Not sure, what you want to do here